### ISSUE TITLE
KG Linkage Problem

### ISSUE THREAD

**Issue by erictopi** [RUCAIBox/RecSysDatasets#130]
Am I missing something or is the link file between the movie IDs and the freebase IDs for the MovieLens dataset incorrect?
For example: Titanic has the ID **313** and is linked to the freebase ID **m.0b8l01** in the link file. If I look up this freebase ID on wikidata I get the movie **The Swan Princess**. For me, the wrong linkage wouldn't be a problem if it was consistant with the knowledge graph. But when I check **m.0b8l01** in the kg file, I get the ID **m.0ccv3s**, which is the director **Richard Rich** who is indeed the director of **The Swan Princess** and has nothing to do with the movie **Titanic**.

Would be great if someone could check if I'm right or explain to me what I'm getting wrong

Cheers

**Comment by baumanno** [comment]
Hi @erictopi, I know it's been a while, but could you perhaps share your insights on this? I'm running into the same phenomenon, and would like to know what's happening and how I could fix up the `*.link` file so that it establishes the correct linking between the item-id and freebase identifier.

**Comment by erictopi** [comment]
@baumanno as far as I remember, I ended up using a different Movielens dataset (I think in my case it was the 1m dataset) that had the correct IDs. I then prepared it with the conversion tools for the recbole framework and used subsamples for my experiments.

**Comment by baumanno** [comment]
TL; DR: if you intend to build a KG from the data published on Google Drive and Baidu, make sure you're using `ml-1m` (or larger, but I checked 1M only) as a base.

@erictopi, you're spot on, thanks for the hint. In my case, I was using `ml-100k`, for which the mapping against the KG data on Google Drive is incorrect. The `.link` file created by `add_knowledge.py` will contain a false mapping of movielens-ID to FreeBase-ID ~for any other dataset except `ml-1m`~ for the `ml-100k` dataset. 

This should be mentioned somewhere, as the docs suggest the conversion tools would work with ~any of the movielens subsets~ the `ml-100k` subset.

Note that larger sets such as 20M may work as well.
